src/command.cpp

```diff
@@ -539,21 +487,25 @@ QoreHashNode * command::fetch_row(ExceptionSink* xsink, const Placeholders *ph)
    return h;
 }
 
-AbstractQoreNode *command::read_rows(const Placeholders *ph, ExceptionSink* xsink) {
+AbstractQoreNode *command::read_rows(const Placeholders *ph, ExceptionSink* xsink, int max_rows) {
    if (ensure_colinfo(xsink)) return 0;
 
    ReferenceHolder<AbstractQoreNode> rv(xsink);
-   QoreListNode *l = 0;
+   QoreListNode *l = nullptr;
    while (fetch_row_into_buffers(xsink)) {
       ReferenceHolder<QoreHashNode> h(output_buffers_to_hash(ph, xsink), xsink);
-      if (*xsink) return 0;
+      if (*xsink) return nullptr;
       if (rv) {
 	 if (!l) {
 	    ReferenceHolder<QoreListNode> lholder(new QoreListNode, xsink);
 	    l = *lholder;
 	    l->push(rv.release());
 	    rv = lholder.release();
 	 }
+         if (max_rows > 0 && l->size() == (size_t)max_rows) {
+            xsink->raiseException("DBI-SELECT-ROW-ERROR", "SQL passed to selectRow() returned more than 1 row");
```

Comment:
Either `max_rows` should be changed to `maxOneRow` and the condition here accordingly, or the error should be fixed since `max_rows` could be set to different values than 1.

Reply:
good point - resolved